psalm.xml

```diff
@@ -23,6 +23,24 @@
         <file name="vendor/jetbrains/phpstorm-stubs/sqlsrv/sqlsrv.php" />
     </stubs>
     <issueHandlers>
+        <ConflictingReferenceConstraint>
+            <errorLevel type="suppress">
+                <!--
+                    This one is just too convoluted for Psalm to figure out, by
+                    its author's own admission
+                -->
+                <file name="lib/Doctrine/DBAL/Driver/OCI8/OCI8Statement.php"/>
```

Comment:
IIRC, it must be something like this but more complicated:
```php
function test(string &$i) : void {
    $i = null;
}
```
Since we're raising the bar, I'd rather fix the issues that are not false-positives, and this one is most likely not. I'll try to find some time and solve this before we proceed.

Reply:
Yeah there are 2 methods like this, one can make `$i` a `null` or a `string` the other one only accepts strings. Great if we can rework the code.

Reply:
Interestingly, even if I change the condition to `if ($currentLiteralDelimiter === null)` or add `assert(is_string($currentLiteralDelimiter))` before calling the function that expects a string, the error still persists. I don't see an easy way to rework this w/o complicating the implementation a lot (i.e. instead of using references, return structures and then extract their properties into the current state).

At this point, it looks like a shortcoming of Psalm rather than a real issue. Unlike the example in the [article](https://psalm.dev/docs/running_psalm/issues/ConflictingReferenceConstraint/), the null value is only possible in the branch that accepts null.

Reply:
Yeah I tried this too, then I asked for support [on Slack and was told I could safely ignore that one](https://app.slack.com/client/T39U3FUKV/C8SFXTD2M/thread/C6JKVFPMG-1587419029.172700)